src/models/CommonModel.ts

```diff
@@ -1,5 +1,12 @@
 import { Logger } from '../utils';
 
+export interface MergingOptions {
+  constrictModels: boolean;
```

Comment:
We might want to clarify this 🤔 
```suggestion
  /**
    * When models are merged, should merging models constrict the `merging to` model? 
    * If false, `required` keyword would no longer be applied from the `merging from` model.
    */
  constrictModels: boolean;
```